### PR TITLE
chore(manifest): add static site

### PR DESCRIPTION
**Pull request by iamhopaul123**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/manifest/static_site.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	return &StaticSite{
+		Workload: Workload{
+			Name: aws.String(name),
+			Type: aws.String(StaticSiteType),
+		},
+		parser: template.New(),
+	}
+}
+
+func (s StaticSite) applyEnv(envName string) (workloadManifest, error) {
```
Comment (efekarakus) [inline internal/pkg/manifest/static_site.go:48]:
```suggestion
func (s *StaticSite) applyEnv(envName string) (workloadManifest, error) {
```

File: internal/pkg/manifest/validate.go [hunk 2]
```diff
@@ -533,6 +533,31 @@ func (s ScheduledJobConfig) validate() error {
...
+	if err = s.Workload.validate(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s StaticSiteConfig) validate() error {
+	for _, fileupload := range s.FileUploads {
+		if err := fileupload.validate(); err != nil {
+			return fmt.Errorf(`validate "files": %w`, err)
```
Comment (efekarakus) [inline internal/pkg/manifest/validate.go]: nit: can we include the index here

**Review by efekarakus — APPROVED** [review]

File: internal/pkg/manifest/static_site.go (hunk 1 above)
Comment (iamhopaul123) [reply internal/pkg/manifest/static_site.go:48]: reverted to be consistent with the other workloads' applyenv

File: internal/pkg/manifest/static_site.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	s.Environments = nil
+	return &s, nil
+}
+
+func (s *StaticSite) subnets() *SubnetListOrArgs {
+	return nil
+}
+
+func (s *StaticSite) requiredEnvironmentFeatures() []string {
+	return nil
```
Comment (Lou1415926) [inline internal/pkg/manifest/static_site.go:78]: just double check: does it not require any env features or is this a todo?

File: internal/pkg/manifest/static_site_test.go [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+func TestStaticSite_ApplyEnv(t *testing.T) {
+	var ()
+	testCases := map[string]struct {
+		in         *StaticSite
+		envToApply string
+
+		wanted *StaticSite
+	}{
+		"with no existing environments": {
```
Comment (Lou1415926) [inline internal/pkg/manifest/static_site_test.go]: nit:
```suggestion
		"without environment overrides": {
```

File: internal/pkg/manifest/validate.go [hunk 5]
```diff
@@ -533,6 +533,31 @@ func (s ScheduledJobConfig) validate() error {
...
+// validate returns nil if StaticSite is configured correctly.
+func (s StaticSite) validate() error {
+	var err error
+	if err = s.StaticSiteConfig.validate(); err != nil {
+		return err
+	}
+	if err = s.Workload.validate(); err != nil {
+		return err
+	}
+	return nil
```
Comment (Lou1415926) [inline internal/pkg/manifest/validate.go]: nit:
```suggestion
	return s.Workload.validate()
```

**Review by Lou1415926 — APPROVED** [review]

File: internal/pkg/manifest/static_site.go (hunk 3 above)
Comment (iamhopaul123) [reply internal/pkg/manifest/static_site.go:78]: it's just for the interface requirement. I'll add a note there.

**Pull request merged**